rotkehlchen/tests/unit/accounting/evm/test_thegraph.py

```diff
@@ -0,0 +1,142 @@
+    pot = accountant.pots[0]
+    events_accountant = pot.events_accountant
+    thegraph_accountant = ThegraphAccountant(
+        node_inquirer=ethereum_inquirer,
+        msg_aggregator=ethereum_inquirer.database.msg_aggregator,
+    )
+    thegraph_settings = thegraph_accountant.event_settings(pot)
+    events_accountant.event_settings.update(thegraph_settings)
+    events_iterator = iter([EvmEvent(
+        tx_hash=HASH1,
+        sequence_index=359,
+        timestamp=TSMS1,
+        location=Location.ETHEREUM,
+        event_type=HistoryEventType.STAKING,
+        event_subtype=HistoryEventSubType.DEPOSIT_ASSET,
+        asset=A_GRT,
+        balance=Balance(amount=FVal('998.98')),
+        location_label=USER_ADDRESS,
+        notes=f'Delegate 998.98 GRT to indexer {ADDY_INDEXER}',
+        counterparty=CPT_THEGRAPH,
+        address=ADDY_THEGRAPH,
+    ), EvmEvent(
+        tx_hash=HASH1,
+        sequence_index=360,
+        timestamp=TSMS1,
+        location=Location.ETHEREUM,
+        event_type=HistoryEventType.SPEND,
+        event_subtype=HistoryEventSubType.FEE,
+        asset=A_GRT,
+        balance=Balance(amount=FVal('5.02')),
+        location_label=USER_ADDRESS,
+        notes='Burn 5.02 GRT as delegation tax',
+        counterparty=CPT_THEGRAPH,
+        address=ADDY_THEGRAPH,
+    ), EvmEvent(
+        tx_hash=HASH2,
+        sequence_index=208,
+        timestamp=TSMS2,
+        location=Location.ETHEREUM,
+        event_type=HistoryEventType.STAKING,
+        event_subtype=HistoryEventSubType.REMOVE_ASSET,
+        asset=A_GRT,
+        balance=Balance(amount=FVal('1003.70342593701668535')),
+        location_label=USER_ADDRESS,
+        notes=f'Withdraw 1003.70342593701668535 GRT from indexer {ADDY_INDEXER}',
+        counterparty=CPT_THEGRAPH,
+        address=ADDY_THEGRAPH,
+    )])
+    for event in events_iterator:
+        pot.events_accountant.process(event=event, events_iterator=events_iterator)
+
+    expected_events = [
+        ProcessedAccountingEvent(
+            type=AccountingEventType.TRANSACTION_EVENT,
+            notes=f'Delegate 998.98 GRT to indexer {ADDY_INDEXER}',
+            location=Location.ETHEREUM,
+            timestamp=TS1,
+            asset=A_GRT,
+            free_amount=ZERO,
+            taxable_amount=FVal('998.98'),
+            price=Price(ONE),
+            pnl=PNL(taxable=ZERO, free=ZERO),  # Deposits are not taxable
+            cost_basis=None,
+            index=0,
+            extra_data={'tx_hash': HASH1.hex()},  # pylint: disable=no-member
+        ), ProcessedAccountingEvent(
+            type=AccountingEventType.TRANSACTION_EVENT,
+            notes='Burn 5.02 GRT as delegation tax',
+            location=Location.ETHEREUM,
+            timestamp=TS1,
+            asset=A_GRT,
+            free_amount=ZERO,
+            taxable_amount=FVal('5.02'),
+            price=Price(ONE),
+            pnl=PNL(taxable=ZERO, free=ZERO),
```

Comment:
This should have a negative PNL here. Which means loss. Since this 5.02 GRT is lost and with simplified price of the test of `1` it should have `-5.02` loss.